src/N3DataFactory.js

```diff
@@ -215,21 +213,24 @@ export function termFromId(id, factory) {
     return factory.literal(id.substr(1, endPos - 1),
             id[endPos + 1] === '@' ? id.substr(endPos + 2)
                                    : factory.namedNode(id.substr(endPos + 3)));
-  case '<':
-    const components = quadId.exec(id);
-    return factory.quad(
-      termFromId(unescapeQuotes(components[1]), factory),
-      termFromId(unescapeQuotes(components[2]), factory),
-      termFromId(unescapeQuotes(components[3]), factory),
-      components[4] && termFromId(unescapeQuotes(components[4]), factory)
-    );
+  case '[':
+    id = JSON.parse(id);
+    break;
   default:
-    return factory.namedNode(id);
+    if (!nested || !Array.isArray(id)) {
+      return factory.namedNode(id);
+    }
   }
+  return factory.quad(
+    termFromId(id[0], factory, true),
+    termFromId(id[1], factory, true),
+    termFromId(id[2], factory, true),
+    id[3] && termFromId(id[3], factory, true)
+  );
 }
```

Comment:
Questions for the future:
- Should N3.js `Term` internally still rely on `id`?
- (For what cases) do we still need `termFromId`, or is `termToId` enough—and can it just be `.hash` on the thing itself then?

Reply:
> (For what cases) do we still need termFromId, or is termToId enough—and can it just be .hash on the thing itself then?

Any use of the `Store` will still require `termFromId` this since terms can't be recovered from a hash. Using a hash is really only an optimisation for term equality (in which case a hash for the `id` is fine).

Reply:
Oh actually - I think I see more what you're saying is that quads could just use a set of ids/hashes in the internal representation that are then looked up.

Reply:
> Any use of the Store will still require termFromId this since terms can't be recovered from a hash.

But that could be an internal function then indeed!